File: Solution/CSharp/13.RomantoInteger.cs
public class Solution {
    public int RomanCharToInt(char c) {
        switch (c) {
            case 'I':
                return 1;
            case 'V':
                return 5;
            case 'X':
                return 10;
            case 'L':
                return 50;
            case 'C':
                return 100;
            case 'D':
                return 500;
            case 'M':
                return 1000;
            default:
                return 0;
        }
    }

    public int RomanToInt(string s) {
        int length = s.Length;
        int i = length - 1;
        int sum = RomanCharToInt(s[i--]);
        while (i >= 0) {
            if (RomanCharToInt(s[i + 1]) > RomanCharToInt(s[i])) {
                sum -= RomanCharToInt(s[i--]);
            } else {
                sum += RomanCharToInt(s[i--]);
            }
        }
        return sum;
    }

File: Solution/Python/13. Roman to Integer.py
class Solution:
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        roman2Int = {
            'I': 1,
            'V': 5,
            'X': 10,
            'L': 50,
            'C': 100,
            'D': 500,
            'M': 1000,
        }
        i = len(s) - 1
        res = roman2Int[s[i]]
        i -= 1
        while i >= 0:
            if roman2Int[s[i + 1]] > roman2Int[s[i]]:
                res -= roman2Int[s[i]]
            else:
                res += roman2Int[s[i]]
            i -= 1
        return res


solution = Solution()
s = "DCXXI"
res = solution.romanToInt(s)
print(res)
